File: (8 kyu) Reversed sequence/(8 kyu) Reversed sequence.py
# 1 Plain solution
def reverse_seq(n):
    arr = []
    for i in range(n):
        arr.append(n - i)
    return arr

# 2 Optimized solution
def reverse_seq(n):
    return [x for x in range(n, 0, -1)]

# 3 Clever solution
def reverse_seq(n):
    return list(range(n, 0, -1))

# 4 Coding golf
reverse_seq = lambda n: list(range(n, 0, -1))

File: (8 kyu) Reversed sequence/(8 kyu) Reversed sequence.js
// 1 Plain solution
const reverseSeq = n => {
  let res = [];

  for (let i = n; i > 0; i--) {
    res.push(i);
  }
  
  return res;
};

// 2 Clever solution
const reverseSeq = (n) =>
  Array(n)
    .fill()
    .map((x, i) => n - i);

// 3 Coding golf
const reverseSeq = length =>
  Array.from(
    {length},
    () => length--
  )
